java/src/main/java/com/databricks/mlflow/utils/SerializationUtils.java

```diff
@@ -0,0 +1,87 @@
+package com.databricks.mlflow.utils;
+
+import java.io.File;
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+/**
+ * Utilities for serializing and deserializing objects to and from various
+ * persistence formats, such as JSON and YAML
+ */
+public class SerializationUtils {
+    private static final ObjectMapper jsonMapper = new ObjectMapper(new JsonFactory());
+    private static final ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
+
+    /**
+     * Produces a JSON string representation of a Java object
+     *
+     * @return A string in valid JSON format
+     */
+    public static String toJson(Object object) throws JsonProcessingException {
+        return jsonMapper.writeValueAsString(object);
+    }
+
+    /**
+     * Produces a Java object representation of a JSON-formatted string
+     *
+     * @param json A string in valid JSON format
+     */
+    public static <T> T fromJson(String json) throws IOException {
+        return jsonMapper.readValue(json, new TypeReference<T>() {});
```

Comment:
I'm not sure I believe that this actually works for complex types. Do we need to define it, btw? Maybe we could remove the methods of SerializationUtils that are not currently used.